src/cdomains/intDomain.ml

```diff
@@ -2231,7 +2231,7 @@ struct
     let congruence_series a c m =
       let rec next a1 c1 a2 c2 =
         if a2 |: a1 then (a2, c2)
-        else next a2 c2 (a1 %: a2) ((c1 -: c2) *: (a1 /: a2))
+        else next a2 c2 (a1 %: a2) (c1 -: (c2 *: (a1 /: a2)))
```

Comment:
What is the intuition behind this change?

Reply:
There were test failures with `meet`, like it not being even commutative! So I dug up the referenced paper and saw that this was just implemented incorrectly.

Reply:
Ok, I guess it is like that in [Granger89], so I guess it was simply a typo in the implementation.

Since we have also discovered typos in the Granger paper, I'd be even more comfortable if we manage to construct an example where the old result leads to unsoundness.

Should be fairly easy given the output of the unit tests?

Reply:
If I change it back now, the failures are:
```
test `congruences (int): meet assoc` failed on ≥ 1 cases:
(3ℤ, 4ℤ, 1+7ℤ) (after 235 shrink steps)

test `congruences (int): meet comm` failed on ≥ 1 cases:
(2ℤ, 1+29ℤ) (after 145 shrink steps)

test `congruences (int): meet leq` failed on ≥ 1 cases:
(8ℤ, 1+3ℤ) (after 158 shrink steps)
```
It could just be that the difference was in precision, not soundness, because all the abstraction validity properties still pass.